internal/pipe/ko/ko.go

```diff
@@ -0,0 +1,335 @@
+// Package ko implements the pipe interface with the intent of
+// building OCI compliant images with ko.
+package ko
+
+import (
+	stdctx "context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sync"
+
+	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login"
+	"github.com/chrismellard/docker-credential-acr-env/pkg/credhelper"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/authn/github"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/ko/pkg/build"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/google/ko/pkg/publish"
+	"github.com/goreleaser/goreleaser/internal/ids"
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+	"golang.org/x/tools/go/packages"
+)
+
+const chainguardStatic = "cgr.dev/chainguard/static"
+
+var (
+	baseImages     sync.Map
+	amazonKeychain authn.Keychain = authn.NewKeychainFromHelper(ecr.NewECRHelper(ecr.WithLogger(io.Discard)))
+	azureKeychain  authn.Keychain = authn.NewKeychainFromHelper(credhelper.NewACRCredentialsHelper())
+	keychain                      = authn.NewMultiKeychain(
+		amazonKeychain,
+		authn.DefaultKeychain,
+		google.Keychain,
+		github.Keychain,
+		azureKeychain,
+	)
+)
+
+// Pipe that build OCI compliant images with ko.
+type Pipe struct{}
+
+func (Pipe) String() string { return "ko" }
+func (Pipe) Skip(ctx *context.Context) bool {
+	return ctx.SkipKo || len(ctx.Config.Kos) == 0
+}
+
+// Default sets the Pipes defaults.
+func (Pipe) Default(ctx *context.Context) error {
+	ids := ids.New("kos")
+	for i := range ctx.Config.Kos {
+		ko := &ctx.Config.Kos[i]
+		if ko.ID == "" {
+			ko.ID = "default"
+		}
+
+		if ko.Build == "" {
+			ko.Build = ko.ID
+		}
+
+		build, err := findBuild(ctx, *ko)
+		if err != nil {
+			return err
+		}
+
+		if len(ko.Ldflags) == 0 {
+			ko.Ldflags = build.Ldflags
+		}
+
+		if len(ko.Flags) == 0 {
+			ko.Flags = build.Flags
+		}
+
+		if len(ko.Env) == 0 {
+			ko.Env = build.Env
+		}
+
+		if ko.Main == "" {
+			ko.Main = build.Main
+		}
+
+		if ko.WorkingDir == "" {
+			ko.WorkingDir = build.Dir
+		}
+
+		if ko.BaseImage == "" {
+			ko.BaseImage = chainguardStatic
```

Comment:
We should try to reuse whatever const ko defines upstream, in case we ever change the default base image in a future ko release.

Luckily @developer-guy has already started toward that in https://github.com/ko-build/ko/pull/904, which we can reuse here when `ko.BaseImage` is not defined by the user.

Reply:
not merged/released yet, right? we can change this in the future then

Reply:
yep, it is not released yet, it is waiting for merge right now